app/models/issues_data_series.rb

```diff
@@ -1,20 +1,43 @@
 class IssuesDataSeries
-  attr_reader :open_issues, :closed_issues
+  attr_reader :issues, :starts_at, :ends_at
 
-  def initialize(open_issues, closed_issues)
-    @open_issues = open_issues
-    @closed_issues = closed_issues
+  def initialize(issues, starts_at, ends_at)
+    @issues = issues
+    @starts_at = starts_at
+    @ends_at = ends_at
   end
 
   def series
-    [
-      {
-        name: "Open Issues",
-        data: {
-          1.day.ago => open_issues.count + closed_issues.count,
-          0.days.ago => open_issues.count,
+    @series ||= begin
+      milestone_titles.map do |milestone_title|
+        {
+          name: milestone_title || "No milestone",
+          data: milestone_data(milestone_title)
         }
-      }
-    ]
+      end
+    end
+  end
+
+  protected
+
+  def milestone_titles
+    @milestone_titles ||= issues.map{|i| i.milestone.try(:title)}.uniq
```

Comment:
Space missing to the left of {.<br>Space between { and | missing.<br>Space missing inside }.